tensorboard/webapp/widgets/data_table/types.ts

```diff
@@ -42,6 +43,44 @@ export enum ColumnHeaderType {
   CUSTOM = 'CUSTOM',
 }
 
+export enum DomainType {
+  DISCRETE,
+  INTERVAL,
+}
+
+export type DiscreteFilterValues = string[] | number[] | boolean[];
+
+export type DiscreteFilterValue = DiscreteFilterValues[number];
+
+export interface DiscreteFilter {
+  type: DomainType.DISCRETE;
+  includeUndefined: boolean;
+  possibleValues: DiscreteFilterValues;
+  // Subset of `possibleValues`
+  filterValues: DiscreteFilterValues;
+}
+
+export interface DiscreteFilterChange {
+  hparamName: string;
+  includeUndefined: boolean;
+  filterValues: DiscreteFilterValues;
+}
+
+export interface IntervalFilter {
+  type: DomainType.INTERVAL;
+  includeUndefined: boolean;
+  minValue: number;
+  maxValue: number;
+  // Filter values have to be in between min and max values (inclusive).
+  filterLowerValue: number;
+  filterUpperValue: number;
+}
+
+export interface FilterAddedEvent {
```

Comment:
Same, I do not see this type used anywhere in this PR. Can we add it when needed?